lib/schedule.c

```diff
@@ -8,132 +8,76 @@
 
 #include "schedule.h"
 #include "reactor.h"
+#include "reactor_common.h"
+#include "environment.h"
 
-/**
- * Schedule an action to occur with the specified value and time offset
- * with no payload (no value conveyed).
- * See schedule_token(), which this uses, for details.
- *
- * @param action Pointer to an action on the self struct.
- * @param offset The time offset over and above that in the action.
- * @return A handle to the event, or 0 if no event was scheduled, or -1 for error.
- */
-trigger_handle_t lf_schedule(void* action, interval_t offset) {
-    return _lf_schedule_token((lf_action_base_t*)action, offset, NULL);
+#include <assert.h>
+#include <string.h> // Defines memcpy.
+
+trigger_handle_t lf_schedule(lf_action_base_t* action, interval_t offset) {
+    return lf_schedule_token((lf_action_base_t*)action, offset, NULL);
 }
 
-/**
- * Schedule the specified action with an integer value at a later logical
- * time that depends on whether the action is logical or physical and
- * what its parameter values are. This wraps a copy of the integer value
- * in a token. See schedule_token() for more details.
- *
- * @param action The action to be triggered.
- * @param extra_delay Extra offset of the event release above that in the action.
- * @param value The value to send.
- * @return A handle to the event, or 0 if no event was scheduled, or -1 for error.
- */
-trigger_handle_t lf_schedule_int(void* action, interval_t extra_delay, int value) {
-    return _lf_schedule_int((lf_action_base_t*)action, extra_delay, value);
+trigger_handle_t lf_schedule_int(lf_action_base_t* action, interval_t extra_delay, int value) {
+    token_template_t* template = (token_template_t*)action;
+
+    // NOTE: This doesn't acquire the mutex lock in the multithreaded version
+    // until schedule_value is called. This should be OK because the element_size
+    // does not change dynamically.
+    if (template->type.element_size != sizeof(int)) {
+        lf_print_error("Action type is not an integer. element_size is %zu", template->type.element_size);
+        return -1;
+    }
+    int* container = (int*)malloc(sizeof(int));
+    *container = value;
+    return lf_schedule_value(action, extra_delay, container, 1);
```

Comment:
Note to other reviewers -- visually checked that this is the same as the original implementation, just in a different place.

(Anyone can go ahead and mark these as resolved when they are no longer helpful)